pkg/webhook/server.go

```diff
@@ -123,13 +124,23 @@ func (s *Server) Register(path string, hook http.Handler) {
 
 // instrumentedHook adds some instrumentation on top of the given webhook.
 func instrumentedHook(path string, hookRaw http.Handler) http.Handler {
-	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
-		startTS := time.Now()
-		defer func() { metrics.RequestLatency.WithLabelValues(path).Observe(time.Since(startTS).Seconds()) }()
-		hookRaw.ServeHTTP(resp, req)
-
-		// TODO(directxman12): add back in metric about total requests broken down by result?
-	})
+	lbl := prometheus.Labels{"webhook": path}
+
+	lat := metrics.RequestLatency.MustCurryWith(lbl)
+	cnt := metrics.RequestTotal.MustCurryWith(lbl)
+	gge := metrics.RequestInFlight.With(lbl)
+
+	// Initialize the most likely HTTP status codes.
+	cnt.WithLabelValues("200")
```

Comment:
What value does initialization here add?

Reply:
It adds the metric to the `/metrics` endpoint with the count set to `0`.
In my experience this allows the user of the metrics just this little more insight about which data they can work with, (used [prometheus/client_golang/prometheus/promhttp/http.go](https://github.com/prometheus/client_golang/blob/master/prometheus/promhttp/http.go#L240) as my example here)